components/DateTimeInput.tsx

```diff
@@ -9,40 +10,93 @@ const { t } = i18n;
 export interface DateTimeInputProps {
   id?: string;
   label: string;
-  name: string;
+  dateName: string;
   required?: boolean;
   startAt?: string;
   endAt?: string;
+  isInvalid?: boolean;
+  onChange: (name: string, isInvalid: boolean) => void;
 }
 
+class DateTimeInputStore {
+  @observable start = '';
+  @observable end = '';
+
+  @action setStart(value: string) {
+    this.start = value;
+  }
+
+  @action setEnd(value: string) {
+    this.end = value;
+  }
+}
+
+const dateTimeInputStore = new DateTimeInputStore();
+
 export const DateTimeInput: FC<DateTimeInputProps> = ({
   id,
   label,
-  name,
+  dateName,
   required,
   startAt,
   endAt,
-}) => (
-  <Form.Group as={Row} className="mb-3" controlId={id}>
-    <Form.Label column sm={2}>
-      {label}
-    </Form.Label>
-    <Col sm={10}>
-      <InputGroup className="mb-3">
-        <InputGroup.Text>{t('time_range')}</InputGroup.Text>
-        <Form.Control
-          name={`${name}StartedAt`}
-          type="datetime-local"
-          required={required}
-          defaultValue={startAt && formatDate(startAt, 'YYYY-MM-DDTHH:mm:ss')}
-        />
-        <Form.Control
-          name={`${name}EndedAt`}
-          type="datetime-local"
-          required={required}
-          defaultValue={endAt && formatDate(endAt, 'YYYY-MM-DDTHH:mm:ss')}
-        />
-      </InputGroup>
-    </Col>
-  </Form.Group>
-);
+  isInvalid,
+  onChange,
+}) => {
+  const handleInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const { name, value } = event.target;
+
+    if (name === `${dateName}StartedAt`) {
+      dateTimeInputStore.setStart(value);
+    } else {
+      dateTimeInputStore.setEnd(value);
+    }
+
+    const isInvalid = isInvalidDateTime(
+      dateTimeInputStore.start,
+      dateTimeInputStore.end,
+    );
+    onChange(dateName, isInvalid);
+  };
+
+  const isInvalidDateTime = (startAt: string, endAt: string) => {
+    if (startAt && endAt) {
+      const startDate = new Date(startAt);
+      const endDate = new Date(endAt);
+      return startDate > endDate;
+    }
+    return false;
+  };
```

Comment:
```suggestion
  const isValidDateTime = (startAt?: string, endAt?: string) =>
      +new Date(startAt) <= +new Date(endAt);
```
非法日期转为数值是 `NaN`，与任何数比较都是 `false`。

Reply:
我还是写的isInvalid
因为start和end的初始值是`''`，转换一下就会变成`NaN`，那isValid就是`false`，变成isInvalid就是`true`，也就是在没有输入任何日期的情况下，就会报错。
所以我还是写的isInvalid的逻辑